### PR TITLE
Obsolete APIs for next major 

### PR DESCRIPTION
**Pull request by aleksandr-samila**
Restored obsolete APIs which are being removed

### CODE REVIEW

File: src/NServiceBus.Core/ObjectBuilder/ServiceProviderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        public static IEnumerable<object> BuildAll(this IServiceProvider serviceProvider, Type typeToBuild) => serviceProvider.GetServices(typeToBuild);
+
+        [ObsoleteEx(
+            Message = "The Release method is not supported.",
+            TreatAsErrorFromVersion = "8",
+            RemoveInVersion = "9")]
+        public static void Release(this IServiceProvider serviceProvider, object instance) => throw new NotImplementedException();
+
+        [ObsoleteEx(
+            Message = "The BuildAndDispatch method is not supported anymore.",
```
Comment (mikeminutillo) [inline src/NServiceBus.Core/ObjectBuilder/ServiceProviderExtensions.cs:76]: Can we put some details around what to do?

```suggestion
            Message = "The BuildAndDispatch method is not supported anymore. Use `IServiceProvider.GetService(typeToBuild)` and call the action on the returned service.",
```

**Review by mikeminutillo — CHANGES_REQUESTED** [review]
I think we need to move the APIApprovals.cs test file back to it's original location. It's non-standard to have it in the ApprovalFiles folder.

File: src/NServiceBus.Core/ObjectBuilder/ServiceProviderExtensions.cs (hunk 1 above)
Comment (HEskandari) [reply src/NServiceBus.Core/ObjectBuilder/ServiceProviderExtensions.cs]: Updated the message.

**Comment by HEskandari** [comment]
@mikeminutillo the file was incidentally moved. I have moved it back.

**Review by mikeminutillo — APPROVED** [review]

**Pull request merged**